### ISSUE TITLE
Incompatible dependencies break other apps

### ISSUE THREAD

**Issue by nursoda** [nextcloud/server#24995]
### How to use GitHub
* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Have Nextcloud 20 with Mail 1.7.2 and Group Folders 8.2.0 apps installed
2. execute `occ groupfolders:scan <id>`, upon completion, an error is given, see below
3. deactivate mail app and re-scan → no error

### Expected behaviour
no error :) Either Nextcloud main application should make sure that incompatible dependencies cannot interfere. Either by technical means or by requireing app devs to "shade" their dependencies (move dependencies to their own name spaces).

### Actual behaviour
Both, the Group folders app and the Mail app, seem to register the `Symfony\Component\Console\Command\Command` class in incompatible versions. This probably leads to the failure after the main task is finished:

```
$ php occ groupfolders:scan 3
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 1       | 0     | 00:00:00     |
+---------+-------+--------------+
An unhandled exception has been thrown:
TypeError: Return value of "OCA\GroupFolders\Command\Scan::execute()" must be of the type int, "null" returned. in /server/cloud/apps/mail/vendor/symfony/console/Command/Command.php:261
Stack trace:
#0 /server/cloud/core/Command/Base.php(169): Symfony\Component\Console\Command\Command->run()
#1 /server/cloud/apps/mail/vendor/symfony/console/Application.php(920): OC\Core\Command\Base->run()
#2 /server/cloud/apps/mail/vendor/symfony/console/Application.php(266): Symfony\Component\Console\Application->doRunCommand()
#3 /server/cloud/apps/mail/vendor/symfony/console/Application.php(142): Symfony\Component\Console\Application->doRun()
#4 /server/cloud/lib/private/Console/Application.php(215): Symfony\Component\Console\Application->run()
#5 /server/cloud/console.php(100): OC\Console\Application->run()
#6 /server/cloud/occ(11): require_once('/server/clo...')
#7 {main}
```

### Server configuration
**Operating system:** Linux 5.10.4-arch2-1 #1 SMP PREEMPT Fri, 01 Jan 2021 05:29:53 +0000 x86_64
**Web server:** nginx/1.18.0 (fpm-fcgi)
**Database:** mysql 10.5.8
**PHP version:** 7.4.13 – Modules loaded: Core, date, libxml, openssl, pcre, zlib, ctype, dom, fileinfo, filter, hash, json, mbstring, pcntl, SPL, PDO, session, posix, readline, Reflection, standard, SimpleXML, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, bcmath, curl, exif, gd, gmp, iconv, intl, mysqli, pdo_mysql, zip, apcu, igbinary, imagick, mailparse, redis, Zend OPcache
**Nextcloud version:** 20.0.4.0
**Updated from an older Nextcloud/ownCloud or fresh install:** Updated (many times ;) )
**Where did you install Nextcloud from:** nextcloud.com
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - admin_audit: 1.10.0
  - apporder: 0.11.0
  - audioplayer: 3.0.0
  - bookmarks: 4.0.5
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.3
  - camerarawpreviews: 0.7.10
  - checksum: 1.0.1
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.3
  - contactsinteraction: 1.1.0
  - cospend: 1.2.7
  - dav: 1.16.2
  - deck: 1.2.3
  - documentserver_community: 0.1.8
  - epubreader: 1.4.5
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_accesscontrol: 1.10.1
  - files_automatedtagging: 1.10.1
  - files_frommail: 0.4.0
  - files_lock: 20.0.0
  - files_mindmap: 0.0.24
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.1
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - forms: 2.1.0
  - groupfolders: 8.2.0
  - impersonate: 1.7.0
  - issuetemplate: 0.7.0
  - keeweb: 0.6.4
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - mail: 1.7.2
  - maps: 0.1.8
  - notes: 4.0.2
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - onlyoffice: 6.2.0
  - password_policy: 1.10.1
  - phonetrack: 0.6.5
  - photos: 1.2.1
  - polls: 1.6.2
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - rainloop: 7.0.3
  - ransomware_detection: 0.10.0
  - ransomware_protection: 1.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - spreed: 10.0.4
  - systemtags: 1.10.0
  - tasks: 0.13.6
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_admin: 3.0.0
  - twofactor_backupcodes: 1.9.0
  - twofactor_email: 2.0.0
  - twofactor_gateway: 0.17.0
  - twofactor_nextcloud_notification: 3.1.0
  - twofactor_totp: 5.0.0
  - twofactor_u2f: 6.0.0
  - updatenotification: 1.10.0
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - cms_pico
  - dashboard
  - encryption
  - files_external
  - firstrunwizard
  - nextcloud_announcements
  - recommendations
  - richdocuments
  - richdocumentscode
  - support
  - survey_client
  - user_ldap
  - user_status
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "skeletondirectory": "\/server\/file\/seyfarth_cloud\/_skeleton",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "cloud.seyfarth.de",
            "1": "test.seyfarth.de",
            "2": "cloud.datenschutz-individuell.de",
            "3": "stage.datenschutz-individuell.de",
            "4": "cloud.selbstbestimmt-digital.de",
            "6": "stage.selbstbestimmt-digital.de",
            "7": "cloud.suevia-ka.de",
            "8": "cloud.owncube.com"
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.4.0",
        "overwrite.cli.url": "https:\/\/cloud.seyfarth.de",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "dbindex": 2,
            "timeout": 1.5
        },
        "logtimezone": "Europe\/Berlin",
        "trashbin_retention_obligation": "auto,30",
        "versions_retention_obligation": "auto,366",
        "default_language": "de",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "updater.release.channel": "stable",
        "theme": "",
        "loglevel": 2,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "simpleSignUpLink.shown": false,
        "maintenance": false,
        "twofactor_enforced": "true",
        "twofactor_enforced_groups": [
            "admin",
            "Eltern"
        ],
        "twofactor_enforced_excluded_groups": [],
        "app_install_overwrite": [
            "files_frommail"
        ]
    }
}
```
</details>

**Comment by nursoda** [comment]
On my NC21 beta test instance I just had this – unsure if these app calls are expected behavior or wrong also. Note the **apps/news** and **apps/bookmarks** strings in the trace:

```text
php occ --verbose user:delete someuser

In MissingNamedParameter.php line 17:
                                                                   
  [Doctrine\DBAL\ArrayParameters\Exception\MissingNamedParameter]  
  Named parameter "user_id" does not have a bound value.           
                                                                   

Exception trace:
  at /server/test/3rdparty/doctrine/dbal/src/ArrayParameters/Exception/MissingNamedParameter.php:17
 Doctrine\DBAL\ArrayParameters\Exception\MissingNamedParameter::new() at /server/test/3rdparty/doctrine/dbal/src/ExpandArrayParameters.php:64
 Doctrine\DBAL\ExpandArrayParameters->acceptNamedParameter() at /server/test/3rdparty/doctrine/dbal/src/SQL/Parser.php:79
 Doctrine\DBAL\SQL\Parser::Doctrine\DBAL\SQL\{closure}() at /server/test/3rdparty/doctrine/dbal/src/SQL/Parser.php:96
 Doctrine\DBAL\SQL\Parser->parse() at /server/test/3rdparty/doctrine/dbal/src/Connection.php:1692
 Doctrine\DBAL\Connection->expandArrayParameters() at /server/test/3rdparty/doctrine/dbal/src/Connection.php:1024
 Doctrine\DBAL\Connection->executeQuery() at /server/test/lib/private/DB/Connection.php:221
 OC\DB\Connection->executeQuery() at /server/test/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php:210
 Doctrine\DBAL\Query\QueryBuilder->execute() at /server/test/lib/private/DB/QueryBuilder/QueryBuilder.php:256
 OC\DB\QueryBuilder\QueryBuilder->execute() at /server/test/lib/public/AppFramework/Db/QBMapper.php:320
 OCP\AppFramework\Db\QBMapper->findEntities() at /server/test/apps/news/lib/Db/FolderMapperV2.php:57
 OCA\News\Db\FolderMapperV2->findAllFromUser() at /server/test/apps/news/lib/Service/FolderServiceV2.php:63
 OCA\News\Service\FolderServiceV2->findAllForUser() at /server/test/apps/news/lib/Service/FolderServiceV2.php:227
 OCA\News\Service\FolderServiceV2->deleteUser() at /server/test/apps/news/lib/Hooks/UserDeleteHook.php:42
 OCA\News\Hooks\UserDeleteHook->handle() at /server/test/lib/private/EventDispatcher/ServiceEventListener.php:76
 OC\EventDispatcher\ServiceEventListener->__invoke() at /server/test/3rdparty/symfony/event-dispatcher/EventDispatcher.php:251
 Symfony\Component\EventDispatcher\EventDispatcher->callListeners() at /server/test/3rdparty/symfony/event-dispatcher/EventDispatcher.php:73
 Symfony\Component\EventDispatcher\EventDispatcher->dispatch() at /server/test/lib/private/EventDispatcher/EventDispatcher.php:86
 OC\EventDispatcher\EventDispatcher->dispatch() at /server/test/lib/private/EventDispatcher/EventDispatcher.php:98
 OC\EventDispatcher\EventDispatcher->dispatchTyped() at /server/test/lib/private/User/User.php:223
 OC\User\User->delete() at /server/test/core/Command/User/Delete.php:65
 OC\Core\Command\User\Delete->execute() at /server/test/apps/bookmarks/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /server/test/apps/bookmarks/vendor/symfony/console/Application.php:1009
 Symfony\Component\Console\Application->doRunCommand() at /server/test/apps/bookmarks/vendor/symfony/console/Application.php:273
 Symfony\Component\Console\Application->doRun() at /server/test/apps/bookmarks/vendor/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /server/test/lib/private/Console/Application.php:215
 OC\Console\Application->run() at /server/test/console.php:100
 require_once() at /server/test/occ:11

user:delete <uid>

$ php occ app:remove bookmarks
bookmarks disabled
bookmarks 4.0.5 removed

$ php occ app:remove news
news disabled
news 15.1.1 removed

$ php occ --verbose user:delete someuser
The specified user was deleted
```

**Comment by nfodor** [comment]
This where the issue is reported i guess: https://source.dussan.org/mirrors/nextcloud/commit/3e6dd86ee4d22b2cdb7f8403b3939c7839698c48

**Comment by nursoda** [comment]
Yup, the signing issue is fixed in 21beta6.

**Comment by escribana** [comment]
The signing issue reappeared here in 21beta7!! I tried to switch from 21beta3 to 21beta7 on command line (updater.phar) and got the InvalidSignatureException for the core and ALL apps. :-/

**Comment by nursoda** [comment]
Despite of you warning I dared to update my three test instances from 21b6 to 21b7. The update itself seems to work but integrity check fails:
<details>
<summary>Output of SERVER/settings/integrity/failed</summary>

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- notifications
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- oauth2
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- theming
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- updatenotification
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_pdfviewer
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- privacy
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- user_status
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- contactsinteraction
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- accessibility
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_videoplayer
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- spreed
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- cloud_federation_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- comments
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- ransomware_protection
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- admin_audit
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- systemtags
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- sharebymail
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- survey_client
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- photos
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_sharing
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- encryption
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- contacts
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- recommendations
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_rightclick
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- federatedfilesharing
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- twofactor_backupcodes
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- nextcloud_announcements
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- tasks
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_mindmap
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- notes
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- federation
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_versions
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- dav
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- provisioning_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_external
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- dashboard
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- lookup_server_connector
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- user_ldap
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- logreader
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- weather_status
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- viewer
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- serverinfo
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- files_trashbin
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- text
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- firstrunwizard
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- support
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- forms
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- onlyoffice
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- settings
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- password_policy
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- workflowengine
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.
- activity
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Certificate is not valid.

Raw output
==========
Array
(
    [core] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [notifications] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [oauth2] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [theming] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [updatenotification] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_pdfviewer] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [privacy] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [user_status] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [contactsinteraction] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [accessibility] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_videoplayer] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [spreed] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [cloud_federation_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [comments] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [ransomware_protection] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [admin_audit] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [systemtags] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [sharebymail] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [survey_client] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [photos] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_sharing] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [encryption] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [contacts] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [recommendations] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_rightclick] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [federatedfilesharing] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [twofactor_backupcodes] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [nextcloud_announcements] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [tasks] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_mindmap] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [notes] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [federation] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_versions] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [dav] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [provisioning_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_external] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [dashboard] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [lookup_server_connector] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [user_ldap] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [logreader] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [weather_status] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [viewer] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [serverinfo] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [files_trashbin] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [text] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [firstrunwizard] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [support] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [forms] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [onlyoffice] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [settings] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [password_policy] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [workflowengine] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

    [activity] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Certificate is not valid.
                )

        )

)
```
</details>

**Comment by nursoda** [comment]
Just wanted to install todays app updates (twofactor_u2f and twofactor_totp) but when I am using the web interface (/settings/apps/updates-Page) I am just bounced back to that page and when I use OCC, I just get
```
$ php occ app:update twofactor_u2f
twofactor_u2f new version available: 6.1.0
twofactor_u2f couldn't be updated
```

**Comment by nursoda** [comment]
Today, on 21.0.0.beta8 and with previous issues apparently fixed, there seemed to be a similar with the news app, but closer inspection reveals a [different cause](https://github.com/nextcloud/news/issues/1107):

```
$ php7 occ app:update news
news new version available: 15.2.0
news couldn't be updated
$ php7 occ -v app:remove news
news disabled
news 15.1.1 removed
$ php7 occ -v app:install news
Error: Could not extract app news: Out-of-path file extraction
 {/tmp/oc_tmp_GtKz9W-folder/news/vendor/bin/phpcs --> ../squizlabs/php_codesniffer/bin/phpcs}
```

**Comment by szaimen** [comment]
Sry but this is with the current architecture not solvable in the server repository, afaik. 
Please report the specific problem with certain apps to the applicable repositories. 
Thank you!
